pkg/kernelmonitor/plugins/types/plugin.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"io"
+
+	"k8s.io/node-problem-detector/pkg/kernelmonitor/types"
+)
+
+// Plugin is the interface of a log parsing plugin.
+type Plugin interface {
```

Comment:
I prefer the later too. kmsg has different read logical, which could be regarded as a `new` kernel log watcher, but not suitable for current buffer approach plugin.